src/containers/reports/ReportsSagas.js

```diff
@@ -244,41 +271,176 @@ function* getReportWatcher() :Generator<*, *, *> {
   yield takeEvery(GET_REPORT, getReportWorker);
 }
 
-
 /*
  *
- * ReportsActions.getReports()
+ * ReportsActions.getReportsByDateRange()
  *
  */
 
-function* getReportsWorker(action :SequenceAction) :Generator<*, *, *> {
+function* getReportsByDateRangeWorker(action :SequenceAction) :Generator<*, *, *> {
+  try {
+    const { value } = action;
+    if (!isDefined(value)) throw ERR_ACTION_VALUE_NOT_DEFINED;
+    if (!Map.isMap(value)) throw ERR_ACTION_VALUE_TYPE;
 
-  const { id, value } = action;
-  if (value === null || value === undefined) {
-    yield put(getReports.failure(id, ERR_ACTION_VALUE_NOT_DEFINED));
-    return;
-  }
+    yield put(getReportsByDateRange.request(action.id));
 
-  const entitySetId :string = value.entitySetId;
+    const edm :Map<*, *> = yield select(state => state.get('edm'));
+    const app = yield select(state => state.get('app', Map()));
 
-  try {
-    yield put(getReports.request(action.id));
-    const response = yield call(getEntitySetDataWorker, getEntitySetData({ entitySetId }));
-    if (response.error) throw response.error;
-    yield put(getReports.success(action.id, response.data));
+    const entitySetId :UUID = getReportESId(app);
+    const peopleESID :UUID = getPeopleESId(app);
+    const appearsInESID :UUID = getAppearsInESId(app);
+    const reportedESID :UUID = getReportedESId(app);
+    const staffESID :UUID = getStaffESId(app);
+
+    const datetimePTID :UUID = edm.getIn(['fqnToIdMap', FQN.DATE_TIME_OCCURRED_FQN]);
+    const entitySetSize = yield call(DataApi.getEntitySetSize, entitySetId);
```

Comment:
Is this absolutely necessary? Last I talked with Matthew, this requests is actually really heavy on the backend, and we might want to avoid it if it's not absolutely necessary.

As an alternative until we implement paging on search, you can set `maxHits` to `10000`:
https://github.com/openlattice/api/blob/develop/src/main/java/com/openlattice/search/SearchApi.java#L77